src/shell/lp_frontend.cpp

```diff
@@ -16,14 +16,21 @@ Copyright (c) 2016 Microsoft Corporation
 #include "util/gparams.h"
 #include <signal.h>
 #include "smt/params/smt_params_helper.hpp"
+#ifndef SINGLE_THREAD
```

Comment:
this is the same pattern in the other frontend files. 
It will be better to coalesce the differences.

Reply:
I can create a special `util/thread.h` header with `ifndef` in it.